File: Backend/StudentManagementSystem/src/main/java/com/ManagementApplication/StudentManagementSystem/service/AttendanceService.java
package com.ManagementApplication.StudentManagementSystem.service;

import com.ManagementApplication.StudentManagementSystem.dto.AttendanceDto;
import java.time.LocalDate;
import java.util.List;

public interface AttendanceService {
    List<AttendanceDto> getAllAttendanceRecords();

    AttendanceDto getAttendanceById(Long id);

    List<AttendanceDto> getAttendanceByStudentId(Long studentId);

    List<AttendanceDto> getAttendanceByCourseId(Long courseId);

    List<AttendanceDto> getAttendanceByDate(String date);

    List<AttendanceDto> getAttendanceByStudentIdAndDate(Long studentId, LocalDate date);

    List<AttendanceDto> getAttendanceByCourseIdAndDate(Long courseId, LocalDate date);

    AttendanceDto createAttendance(AttendanceDto attendanceDto);

    AttendanceDto updateAttendanceById(Long id, AttendanceDto attendanceDto);

    void deleteAttendanceById(Long id);
}

File: frontend/school-management-frontend/src/services/AttendanceService.js
import axios from "axios";

const ATTENDANCE_API_BASE_URL = 'http://localhost:8080/api/attendance';

class AttendanceService {
    getAllAttendance() {
        return axios.get(ATTENDANCE_API_BASE_URL);
    }

    getAttendanceById(id) {
        return axios.get(ATTENDANCE_API_BASE_URL + '/' + id);
    }

    createAttendance(attendance) {
        return axios.post(ATTENDANCE_API_BASE_URL, attendance);
    }

    updateAttendance(id, attendance) {
        return axios.put(ATTENDANCE_API_BASE_URL + '/' + id, attendance);
    }

    deleteAttendance(id) {
        return axios.delete(ATTENDANCE_API_BASE_URL + '/' + id);
    }

    getAttendanceByStudentId(studentId) {
        return axios.get(ATTENDANCE_API_BASE_URL + '/student/' + studentId);
    }

    getAttendanceByCourseId(courseId) {
        return axios.get(ATTENDANCE_API_BASE_URL + '/course/' + courseId);
    }

    getAttendanceByDate(date) {
        return axios.get(ATTENDANCE_API_BASE_URL + '/date/' + date);
    }
}

export default new AttendanceService(); 